extmod/vfs_fat_misc.c

```diff
@@ -66,10 +66,10 @@ STATIC mp_obj_t mp_vfs_fat_ilistdir_it_iternext(mp_obj_t self_in) {
         }
         if (fno.fattrib & AM_DIR) {
             // dir
-            t->items[1] = MP_OBJ_NEW_SMALL_INT(0x4000); // S_IFDIR
```

Comment:
But we were in the process of discussing this! I was waiting for your turn in considering of possible problems with not using DT_* types, and if none, then using stat's values makes good sense!